File: API-Biblioteca-DevEficiente-CDD/src/main/java/io/github/monthalcantara/bibliotecaCDD/usuario/UsuarioController.java
package io.github.monthalcantara.bibliotecaCDD.usuario;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;

@RestController
@RequestMapping("/v1/usuarios")
public class UsuarioController {

    private Logger logger = LoggerFactory.getLogger(UsuarioController.class);
    private UsuarioRepository usuarioRepository;

    public UsuarioController(UsuarioRepository usuarioRepository) {
        this.usuarioRepository = usuarioRepository;
    }

    @PostMapping
    public ResponseEntity cadastraUsuario(@RequestBody @Valid NovoUsuarioRequest novoUsuarioRequest){
        logger.info("Recebida requisição para criação de novo usuario");

        Usuario usuario = novoUsuarioRequest.toModel();

        logger.info("Realizada conversão de Usuariorequest para  usuario");

        usuario = usuarioRepository.save(usuario);

        logger.info("Usuario salvo no banco de dados");

        UsuarioResponse usuarioResponse = new UsuarioResponse(usuario);

        logger.info("Realizada conversão de Usuario salvo para  UsuarioResponse");

        return new ResponseEntity(usuarioResponse, HttpStatus.CREATED);
    }
}

File: Kotlin/API-Casa_Do_Codigo_Kotlin/src/main/kotlin/io/github/monthalcantara/casadocodigo/controller/usuario/UsuarioController.kt
package io.github.monthalcantara.casadocodigo.controller.usuario

import io.github.monthalcantara.casadocodigo.dto.request.usuario.NovoUsuarioRequest
import io.github.monthalcantara.casadocodigo.dto.response.UsuarioResponse
import io.github.monthalcantara.casadocodigo.model.Usuario
import io.github.monthalcantara.casadocodigo.service.UsuarioService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import javax.validation.Valid

@RestController
@RequestMapping("/usuarios")
class UsuarioController(val usuarioService: UsuarioService) {

    @PostMapping
    fun cadastra(@Valid @RequestBody usuariorequest: NovoUsuarioRequest): ResponseEntity<UsuarioResponse> {
        val usuarioEntity = usuariorequest.toModel()

        val (id, username,_,_) = usuarioService.salva(usuarioEntity)
        return ResponseEntity(UsuarioResponse(id, username), HttpStatus.CREATED)
    }
}